### PR TITLE
enforce timeout earlier on connectionClient

### PR DESCRIPTION
**Pull request by mmaxim**

### CODE REVIEW

File: rpc/connection.go [hunk 1]
```diff
@@ -627,8 +627,13 @@ func (c *Connection) connect(ctx context.Context) error {
...
-func (c *Connection) DoCommand(ctx context.Context, name string,
+func (c *Connection) DoCommand(ctx context.Context, name string, timeout time.Duration,
 	rpcFunc func(GenericClient) error) error {
+	if timeout > 0 {
+		var timeoutCancel context.CancelFunc
+		ctx, timeoutCancel = context.WithTimeout(ctx, timeout)
+		defer timeoutCancel()
+	}
 	for {
 		if c.initialReconnectBackoffWindow != nil && isWithFireNow(ctx) {
```
Comment (joshblum) [inline rpc/connection.go]: check for ctx.Done here? or does some function within this do that?

Comment (mmaxim) [reply rpc/connection.go]: Sure

Comment (mmaxim) [reply rpc/connection.go]: Actually don't think we need to, the relevant places check it

Comment (joshblum) [reply rpc/connection.go]: great

**Pull request merged**